src/enum_traits.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#ifndef CATA_ENUM_TRAITS_H
+#define CATA_ENUM_TRAITS_H
+
+template<typename E>
+struct enum_traits;
+
+namespace enum_traits_detail
+{
+
+template<typename E>
+using last_type = typename std::decay<decltype( enum_traits<E>::last )>::type;
+
+} // namespace detail
```

Comment:
Very noisy output in the [logs](https://travis-ci.org/CleverRaven/Cataclysm-DDA/jobs/572542661).
```suggestion
} // namespace enum_traits_detail
```

Reply:
Fixed.